spec/support/shared_examples/validators/shared_examples_for_advocate_claim_validators.rb

```diff
@@ -108,30 +108,38 @@
 end
 
 RSpec.shared_examples 'common defendant uplift fees aggregation validation' do |options|
-  let(:fee_type) { Fee::MiscFeeType.find_by(unique_code: options[:unique_codes][0]) }
-  let(:fee_type_2) { Fee::MiscFeeType.find_by(unique_code: options[:unique_codes][2]) }
-  let(:uplift_fee_type) { Fee::MiscFeeType.find_by(unique_code: options[:unique_codes][1]) }
-  let(:uplift_fee_type_2) { Fee::MiscFeeType.find_by(unique_code: options[:unique_codes][3]) }
+  case options[:claim_type]
```

Comment:
This is a complex area but all this conditional logic implies it should not be shared  (should not need it if behaviour is supposed to be same).

However, I think we can get rid of most if not all conditional logic and just use fee types that are valid/can-exist on any of the three claim types as all the test it trying to ensure is that uplifts are valid and invalid appropriately.

We can talk about this tomorrow but i reckon we should:

1. remove behaviour that is not shared between all three claims types to there or contexts or shared examples  - see my other comment below.
2. only use fee types here that are claimable on all three claim types, such as these:
```
Confiscation hearings (whole day),DTW,MIDTW
Confiscation hearings (whole day uplift),DWU,MIDWU
Deferred sentence hearings,DSE,MIDSE,
Deferred sentence hearings uplift,DSU,MIDSU
```
and get rid of the case statement and options entirely

3. For clarity rename `uplift_fee_type_1` to `uplift_for_fee_type_1` & rename `uplift_fee_type_2` to `uplift_for_fee_type_2`

Reply:
1. 👍 Have created 3 new shared examples contexts; shared_context 'seeded fee schemes', shared_examples 'common defendant uplift fees aggregation validation' and shared_examples 'common defendant basic fees aggregation validation' .

2. 👍 Have amended to use suggested fee types that are claimable on all three claim types.

3. I have reverted to the original fee_type naming (midtw, midse...) for clarity.